src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubWebhookListenerImpl.java

```diff
@@ -40,20 +43,25 @@
 
     @Override
     public void onPushRepositoryChanged(String pusherName, final GitHubRepositoryName changedRepository) {
-        ACL.impersonate(ACL.SYSTEM, new Runnable() {
+        // Delaying the indexing for some seconds to avoid GitHub cache
+        Timer.get().schedule(new Runnable() {
             @Override public void run() {
-                for (final SCMSourceOwner owner : SCMSourceOwners.all()) {
-                    for (SCMSource source : owner.getSCMSources()) {
-                        if (source instanceof GitHubSCMSource) {
-                            GitHubSCMSource gitHubSCMSource = (GitHubSCMSource) source;
-                            if (gitHubSCMSource.getRepoOwner().equals(changedRepository.getUserName()) &&
-                                    gitHubSCMSource.getRepository().equals(changedRepository.getRepositoryName())) {
-                                owner.onSCMSourceUpdated(gitHubSCMSource);
+                ACL.impersonate(ACL.SYSTEM, new Runnable() {
+                    @Override public void run() {
+                        for (final SCMSourceOwner owner : SCMSourceOwners.all()) {
+                            for (SCMSource source : owner.getSCMSources()) {
+                                if (source instanceof GitHubSCMSource) {
+                                    GitHubSCMSource gitHubSCMSource = (GitHubSCMSource) source;
+                                    if (gitHubSCMSource.getRepoOwner().equals(changedRepository.getUserName()) &&
+                                            gitHubSCMSource.getRepository().equals(changedRepository.getRepositoryName())) {
+                                        owner.onSCMSourceUpdated(gitHubSCMSource);
+                                    }
+                                }
                             }
                         }
                     }
-                }
+                });
             }
-        });
+        }, 10, TimeUnit.SECONDS);
```

Comment:
@amuniz Could you add a comment with the Jira issue? It is easy to forget why we did this and difficult to understand for other.